### PR TITLE
lang/agda-mode: New layer

### PR DESCRIPTION
**Pull request by ocharles**

### CODE REVIEW

**Comment by ocharles** [comment]
The only thing I couldn't quite work out was how to specify the faces. I ended up using `custom-set-faces`, as it's the only one that seems to completely replace definitions, which is what I want.

**Comment by trishume** [comment]
Might want to check out some of the configs @chrisbarrett has as well:
https://github.com/chrisbarrett/.emacs.d/blob/master/private%2Fcb-agda%2Fextensions.el

**Comment by ocharles** [comment]
The remapping stuff there looks interesting, but it relies on Chris' custom config stuff. With the other stuff, I don't really know about that. My preference would be to merge as is and let others build on my work :)

File: contrib/lang/agda/extensions.el [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+   '(agda2-highlight-inductive-constructor-face ((t (:inherit font-lock-function-name-face))))
+   '(agda2-highlight-keyword-face ((t (:inherit font-lock-keyword-face))))
+   '(agda2-highlight-module-face ((t (:inherit font-lock-constant-face))))
+   '(agda2-highlight-number-face ((t nil)))
+   '(agda2-highlight-postulate-face ((t (:inherit font-lock-type-face))))
+   '(agda2-highlight-primitive-type-face ((t (:inherit font-lock-type-face))))
+   '(agda2-highlight-record-face ((t (:inherit font-lock-type-face)))))
+
+  (load-file (let ((coding-system-for-read 'utf-8))
+               (shell-command-to-string "agda-mode locate")))
```
Comment (syl20bnr) [inline contrib/lang/agda/extensions.el]: @ocharles We need a better way to locate the directory and then add this directory to load path. Does agda come with any env. variable setup or something ?

Comment (ocharles) [inline contrib/lang/agda/extensions.el]: What type of environment variable are you thinking of? Not one that is going to be in the set of environment variables Emacs would see. The above is the recommended way to use Agda, so it doesn't seem too bad here - people who enable the layer likely already have Agda.

Comment (chrisbarrett) [inline contrib/lang/agda/extensions.el]: It's weird, but yeah, that is apparently the recommended way of loading agda-mode, since it's not on elpa.

**Comment by ocharles** [comment]
Could this be merged? I understand the loading might not be ideal, but I believe it's better than having no layer at all :)

**Comment by syl20bnr** [comment]
Don't worry it is planned to be merged before the release, I try to merge all the old PRs with new layer before the release. I have to find enough time to play a bit with agda before merging this.

**Comment by syl20bnr** [comment]
Thank you ! :+1: 
The layer is now merged ! :-)

I made some improvements listed in this commit: https://github.com/syl20bnr/spacemacs/commit/75c170c

Cherry-picked into `develop`, you can safely delete your branch.

**Comment by ocharles** [comment]
Wow, these improvements look fantastic! Thanks for being such an excellent maintainer, @syl20bnr :heart:

**Pull request closed without merging**